lit_nlp/examples/tydi_demo.py

```diff
@@ -0,0 +1,103 @@
+r"""Example demo loading a T5 model.
+
+To run locally with a small number of examples:
+  python -m lit_nlp.examples.tydi_demo \
+      --alsologtostderr --port=5432 --max_examples=10
+
+To run using the nearest-neighbor lookup index (warning, this will take a while
+to load):
+  python -m lit_nlp.examples.tydi_demo \
+      --alsologtostderr --port=5432 --warm_start 1.0 \
+      --use_indexer --initialize_index --data_dir=/tmp/t5_index
+
+Then navigate to localhost:5432 to access the demo UI.
+"""
+import os
+import sys
+from typing import Optional, Sequence
+
+from absl import app
+from absl import flags
+from absl import logging
+
+from lit_nlp import dev_server
+from lit_nlp import server_flags
+from lit_nlp.components import word_replacer
+from lit_nlp.examples.datasets import summarization
+from lit_nlp.examples.models import tydi
+
+# NOTE: additional flags defined in server_flags.py
+
+FLAGS = flags.FLAGS
+
+FLAGS.set_default("development_demo", True)
+
+_MODELS = flags.DEFINE_list(
+    "models", ["mrm8488/bert-multi-cased-finedtuned-xquad-tydiqa-goldp"],
+    "Models to load")
+_NUM_TO_GEN = flags.DEFINE_integer(
+    "num_to_generate", 4, "Number of generations to produce for each input.")
+
+_TOKEN_TOP_K = flags.DEFINE_integer(
+    "token_top_k", 10, "Rank to which the output distribution is pruned.")
+
+_MAX_EXAMPLES = flags.DEFINE_integer(
+    "max_examples", 1000,
+    "Maximum number of examples to load from each evaluation set. Set to None to load the full set."
+)
+
+_LOAD_BWB = flags.DEFINE_bool(
+    "load_bwb", False,
+    "If true, will load examples from the Billion Word Benchmark dataset. This may download a lot of data the first time you run it, so disable by default for the quick-start example."
+)
+
+
+def get_wsgi_app() -> Optional[dev_server.LitServerType]:
+  FLAGS.set_default("server_type", "external")
+  FLAGS.set_default("demo_mode", True)
+  # Parse flags without calling app.run(main), to avoid conflict with
+  # gunicorn command line flags.
+  unused = flags.FLAGS(sys.argv, known_only=True)
+  return main(unused)
+
+
+def main(argv: Sequence[str]) -> Optional[dev_server.LitServerType]:
+  if len(argv) > 1:
+    raise app.UsageError("Too many command-line arguments.")
+
+  ##
+  # Load models, according to the --models flag.
+  models = {}
+  for model_name_or_path in _MODELS.value:
+    # Ignore path prefix, if using /path/to/<model_name> to load from a
+    # specific directory rather than the default shortcut.
+    model_name = os.path.basename(model_name_or_path)
+    models[model_name] = tydi.TydiModel(
+          model=model_name_or_path,
+          model_name=model_name_or_path,
+          num_to_generate=_NUM_TO_GEN.value,
+          token_top_k=_TOKEN_TOP_K.value,
+          output_attention=False)
```

Comment:
Figured out a few problems originating here (and also with Python just generally being Python)...

We were seeing bugs in the `TyDiModel.predict_minibatch(...)` function when [using `self.model`](https://github.com/PAIR-code/lit/pull/753/files#diff-e66e2ed65042046b3905d9bc4b44fa2c3bb19cb8c07da5a9540002725714fab0R144) because the `model` parameter in this initializer was being passed a string value when it should either be `None` or a `FlaxBertForQuestionAnswering` instance. 

Looking at the [docs for `FlaxBertForQuestionAnswering`](https://huggingface.co/transformers/v4.6.0/model_doc/bert.html#flaxbertforquestionanswering), it also doesn't accept the config parameters being passed in here (e.g., `num_to_generate`, _token_top_k`, `output_attention`), so we can probably simplify this initialization to the following.

```suggestion
    models[model_name] = tydi.TydiModel(model_name=model_name_or_path)
```

I have a few more comments in the `tydi.py` file above related to actually getting predictions from the model.